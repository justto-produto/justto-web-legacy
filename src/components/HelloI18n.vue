<template>
  <p>{{ $t('hello') }}</p>
</template>

<script>
export default {
  name: 'HelloI18n'
}
</script>

<style media="screen">

</style>

<i18n>
{
  "en": {
    "hello": "Hello i18n in SFC!"
  },
  "pt-BR": {
    "hello": "Ol√° i18n in SFC!"
  }
}
</i18n>
