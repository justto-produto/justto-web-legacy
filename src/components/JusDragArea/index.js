import JusDragArea from './JusDragArea.vue'

export { JusDragArea }
