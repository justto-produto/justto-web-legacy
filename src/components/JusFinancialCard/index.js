import JusFinancialCard from './JusFinancialCard'

export { JusFinancialCard }
