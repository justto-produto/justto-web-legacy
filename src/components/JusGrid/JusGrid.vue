<template>
  <div
    :style="{
      gridAutoFlow: flow,
      gridTemplateColumns: `repeat(${columns}, 1fr)`,
      gridTemplateRows: `repeat(${rowsCount}, auto)`,
      gap: toPixel(spacing),
    }"
    class="jus-grid"
  >
    <slot/>
  </div>
</template>

<script>
export default {
  name: 'JusGrid',
  props: {
    flow: {
      type: String,
      default: null,
      validator: val => ['column', 'row'].includes(val),
    },
    rows: {
      type: Number,
      default: null,
    },
    columns: {
      type: Number,
      default: 1,
    },
    spacing: {
      type: Number,
      default: 24,
    },
  },
  computed: {
    rowsCount() {
      return this.rows || 'auto-fit'
    },
  },
  methods: {
    toPixel: num => `${num}px`,
  },
}
</script>

<style lang="scss" scoped>
.jus-grid {
  display: grid;
  width: 100%;
}
</style>
