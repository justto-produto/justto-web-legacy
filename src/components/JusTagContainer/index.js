import JusTagContainer from './JusTagContainer'

export { JusTagContainer }
