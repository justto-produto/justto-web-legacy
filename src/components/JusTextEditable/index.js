import JusTextEditable from './JusTextEditable'

export { JusTextEditable }
