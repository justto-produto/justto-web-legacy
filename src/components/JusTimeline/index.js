import JusTimeline from './JusTimeline'

export { JusTimeline }
