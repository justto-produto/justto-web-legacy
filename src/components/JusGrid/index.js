import JusGrid from './JusGrid'

export { JusGrid }
