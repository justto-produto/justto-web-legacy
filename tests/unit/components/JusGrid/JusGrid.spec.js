import { JusGrid } from '@/components/JusGrid'
import { shallowMount } from '@vue/test-utils'

describe('Components > JusGrid', () => {
  const wrapper = shallowMount(JusGrid)

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
