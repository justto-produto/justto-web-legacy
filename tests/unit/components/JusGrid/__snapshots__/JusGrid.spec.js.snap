// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > JusGrid renders correctly 1`] = `
<div
  class="jus-grid"
  style="grid-template-columns: repeat(1, 1fr); grid-template-rows: repeat(auto-fit, auto); gap: 24px;"
/>
`;
